test/unit/proxy_config_test.rb

```diff
@@ -113,6 +113,17 @@ def test_create_without_service_token
     assert proxy_config.valid?
   end
 
+  test 'missing api backend when service_mesh' do
+    service = FactoryBot.create(:simple_service)
+    proxy = FactoryBot.create(:proxy, service: service)
+    proxy.stubs(:service_mesh_integration?).returns(true)
```

Comment:
Better change it for real in the DB so we do not rely on being the same instance in memory.
```suggestion
    proxy.update!(deployment_option: Service::DeploymentOption.service_mesh)
```

Reply:
It says `deployment_option` is not an attribute so I'm gonna use the stub for now if that's OK